### PR TITLE
add commands for Elytron Audit Logging

### PR DESCRIPTION
**Pull request by jtymel**

### CODE REVIEW

File: commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/audit/AddFileAuditLog.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+public final class AddFileAuditLog implements OnlineCommand {
+
+    private final String name;
+    private final String path;
+    private final String relativeTo;
+    private final Boolean paramSynchronized;
+    private final AuditFormat format;
+    private final boolean replaceExisting;
+
+    public AddFileAuditLog(Builder builder) {
```
Comment (olukas) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/audit/AddFileAuditLog.java]: private

File: commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/audit/AddSyslogAuditLog.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    private final String name;
+    private final String serverAddress;
+    private final Integer port;
+    private final TransportProtocolType transport;
+    private final AuditFormat format;
+    private final String hostName;
+    private final boolean replaceExisting;
+
+    public AddSyslogAuditLog(Builder builder) {
```
Comment (olukas) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/audit/AddSyslogAuditLog.java]: private

File: commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/audit/AddSyslogAuditLog.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            if (serverAddress == null || serverAddress.isEmpty()) {
+                throw new IllegalArgumentException("Server-address must not be null and must have a minimum length of 1 character");
+            }
+            if (port == null) {
+                throw new IllegalArgumentException("Port must not be null");
+            }
+            if (hostName == null || hostName.isEmpty()) {
+                throw new IllegalArgumentException("Host-name must not be null and must have a minimum length of 1 character");
+            }
+
```
Comment (olukas) [inline commands/src/main/java/org/wildfly/extras/creaper/commands/elytron/audit/AddSyslogAuditLog.java:107]: 2 empty lines

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/audit/AddAggregateSecurityEventListenerOnlineTest.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+                ops.exists(TEST_AGGREGATE_SECURITY_EVENT_LISTENER_ADDRESS));
+        client.apply(addAggregateSecurityEventListener2);
+
+        // check whether it was really rewritten
+        checkAttribute(TEST_AGGREGATE_SECURITY_EVENT_LISTENER_ADDRESS, "security-event-listeners[1]",
+                TEST_FILE_AUDIT_LOG_NAME2);
+    }
+
+    @Test(expected = CommandFailedException.class)
+    public void addAggregateSecurityEventListener_notConfiguredSecurityEventListener() throws Exception {
```
Comment (olukas) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/audit/AddAggregateSecurityEventListenerOnlineTest.java]: This does not not test Creaper command. It tests model, I suggest to move it to different testsuite.

File: testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/audit/AddAggregateSecurityEventListenerOnlineTest.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+                = new AddAggregateSecurityEventListener.Builder(TEST_AGGREGATE_SECURITY_EVENT_LISTENER_NAME2)
+                .addSecurityEventListeners(TEST_FILE_AUDIT_LOG_NAME, TEST_SYSLOG_AUDIT_LOG_NAME)
+                .build();
+
+        client.apply(addAggregateSecurityEventListener);
+        client.apply(addAggregateSecurityEventListener2);
+
+        assertTrue("Aggregate security event listener should be created",
+                ops.exists(TEST_AGGREGATE_SECURITY_EVENT_LISTENER_ADDRESS));
+        assertTrue("Second aggregate security event listener should be created",
```
Comment (olukas) [inline testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/audit/AddAggregateSecurityEventListenerOnlineTest.java:110]: It should test whether TEST_AGGREGATE_SECURITY_EVENT_LISTENER_ADDRESS2 exists.

**Review by olukas — CHANGES_REQUESTED** [review]

**Comment by jtymel** [comment]
Thank you @olukas ! The same mistakes again... I swear I'll do it properly once or twice in the future :see_no_evil:

**Pull request merged**